### PR TITLE
feat(icon): support for dynamic type

### PR DESCRIPTION
**Pull request by sean-perkins**
Converts the usage of px units for fonts to rem.

### CODE REVIEW

**Comment by thetaPC** [comment]
> Should this target a different branch instead of `main`?

Yes, it should target "FW-4146" branch instead of main.

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
Can we add a screenshot test of icon being used with the text scaled up? Here's an example you can follow: https://github.com/ionic-team/ionic-framework/pull/27848/files#diff-2796ab1a929613b79f68ba352b11a68994e442070db3a7bbdbd7a1c0ded56c47R9-R26

(Note: previously the team discussed using a linter to ensure rem usage, but as I got more into development I realized some of these components are a bit more in-depth than I initially thought, so to be safe we're going to add a screenshot test)

**Comment by sean-perkins** [comment]
@thetaPC it looks like the `FW-4146` branch is only on Ionic Framework's repository. Are you suggesting we create a similar branch here as well?

**Comment by thetaPC** [comment]
> 

Oops, I misspoke. I though this was on the Framework repo. Disregard my comment.

File: src/components/icon/test/icon.e2e.ts-snapshots/icon-diff-Mobile-Chrome-linux.png
Comment (liamdebeasi) [inline src/components/icon/test/icon.e2e.ts-snapshots/icon-diff-Mobile-Chrome-linux.png:1]: Any idea why this screenshot changed?

**Pull request merged**